modules/Rock.cmake

```diff
@@ -108,9 +108,16 @@ endfunction()
 
 
 ## Main initialization for Rock CMake projects
-macro (rock_init PROJECT_NAME PROJECT_VERSION)
-    project(${PROJECT_NAME})
-    set(PROJECT_VERSION ${PROJECT_VERSION})
+macro (rock_init)
+    if (${ARGC} GREATER 0)
+        message(WARNING "Passing project name and version to rock_init was a misfeature")
+        message(WARNING "of Rock's macros since CMake 3.0. You must call CMake's project()")
+        message(WARNING "at toplevel")
+        set(PROJECT_NAME ${ARG1})
```

Comment:
Why do you set PROJECT_NAME and PROJECT_VERSION if project(...) will do this already, for readability?

Reply:
Artifacts from older revisions. Will remove.